crates/ide-completion/src/tests/special.rs

```diff
@@ -636,3 +644,121 @@ fn bar() -> Bar {
             "#]],
     )
 }
+
+#[test]
+fn completes_fn_in_pub_trait_generated_by_macro() {
+    let mut config = TEST_CONFIG.clone();
+    config.enable_private_editable = false;
```

Comment:
Shall we make it `false` by default in tests?
I've tried locally and there will be failures in only 1 test case `test_visibility_filtering`. Maybe we can enable private editable for this test case?

```rust
        check(
            r#"
//- /lib.rs crate:lib new_source_root:local
pub mod m {
    pub struct A {
        private_field: u32,
        pub pub_field: u32,
        pub(crate) crate_field: u32,
        pub(super) super_field: u32,
    }
}
//- /main.rs crate:main deps:lib new_source_root:local
fn foo(a: lib::m::A) { a.$0 }
"#,
            expect![[r#"
                fd crate_field   u32
                fd private_field u32
                fd pub_field     u32
                fd super_field   u32
            "#]],
        );
```

Reply:
No, I think our default test config should probably be as close to the default config we give users as possible.

Testing this feature specifically seems fine to me here, though making the changed config struct a new local const item here might be nice (or adding a `check_with_private_editable` function), either works.

Reply:
Hi @Veykril , what do you think about this question :)

Reply:
I think the default config we give to users is `enable_private_editable == false`?

Reply:
Related change: https://github.com/rust-lang/rust-analyzer/commit/2a7793d912bafef2d97c0e899262b94efe977543

Reply:
Sure, will do it tomorrow before merging this pr.
(Sorry for the delayed reply...Seems the GitHub pr is not realtime)

Reply:
No problem :)

Reply:
Hi @Veykril, I've made the change. I'd like you to review it again in case I made something wrong :) 
Thanks!